lib/setup.js

```diff
@@ -61,16 +64,15 @@ PouchDB.parseAdapter = function (name, opts) {
   }
 
   adapter = PouchDB.adapters[adapterName];
-  if (adapterName && adapter) {
-    var use_prefix = 'use_prefix' in adapter ? adapter.use_prefix : true;
 
-    return {
-      name: use_prefix ? PouchDB.prefix + name : name,
-      adapter: adapterName
-    };
-  }
+  // if adapter is invalid, then an error will be thrown later
```

Comment:
There IS a test already that confirms that if you do `new PouchDB('foo', {adapter: 'fake'})` then it will throw a useful error.